### PR TITLE
OSOE-118: Update rulesets

### PR DESCRIPTION
**Pull request by sarahelsaig**
OSOE-118

### CODE REVIEW

File: orchardcore.ruleset [hunk 1]
```diff
@@ -2,15 +2,20 @@
...
   </Rules>
+  <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp" RuleNamespace="Microsoft.CodeAnalysis.CSharp">
+    <Rule Id="CS1591" Action="None" />
+  </Rules>
   <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.NetAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.CSharp.NetAnalyzers">
     <Rule Id="CA1019" Action="Hidden" />
     <Rule Id="CA1812" Action="Hidden" />
   </Rules>
   <Rules AnalyzerId="Microsoft.CodeAnalysis.NetAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.NetAnalyzers">
+    <Rule Id="CA1014" Action="None" />
```
Comment (0liver) [inline orchardcore.ruleset:18]: The same code in 4 places - really? 🧷

File: orchardcore.ruleset [hunk 2]
```diff
@@ -2,15 +2,20 @@
 <RuleSet Name="Orchard Core C# rules" ToolsVersion="17.0">
   <Include Path="general.ruleset" Action="Default" />
   <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
+    <Rule Id="CA1014" Action="None" />
     <Rule Id="CA1710" Action="Hidden" />
     <Rule Id="CA1711" Action="Hidden" />
     <Rule Id="CA1716" Action="Hidden" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp" RuleNamespace="Microsoft.CodeAnalysis.CSharp">
+    <Rule Id="CS1591" Action="None" />
```
Comment (0liver) [inline orchardcore.ruleset:11]: I would love for these R# warnings to disappear, but they're still there:

image

Any idea?

**Review by 0liver — APPROVED** [review]

File: .editorconfig [hunk 3]
```diff
@@ -128,6 +128,10 @@ dotnet_diagnostic.IDE0022.severity = warning
 dotnet_diagnostic.IDE0007.severity = warning
 dotnet_diagnostic.IDE0008.severity = none
 
+# We don't want to decorate every public member with XML comments, only public APIs - so turn CS1591 off.
+# CS1591 Missing XML Comment for publicly visible type or member
+dotnet_diagnostic.CS1591.severity = none
+
```
Comment (0liver) [inline .editorconfig:134]: @DAud-IcI Do you agree?

File: orchardcore.ruleset (hunk 2 above)
Comment (0liver) [reply orchardcore.ruleset:11]: I found a solution [here](https://github.com/Lombiq/.NET-Analyzers/pull/62/files#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5bR133).

File: orchardcore.ruleset (hunk 1 above)
Comment (sarahelsaig) [reply orchardcore.ruleset:18]: Yes, it's depressing. (What does safety pin emoji mean here?)

Comment (0liver) [reply orchardcore.ruleset:18]: Being extra safe 😁

Comment (sarahelsaig) [reply orchardcore.ruleset:18]: lol 😹

File: .editorconfig (hunk 3 above)
Comment (sarahelsaig) [reply .editorconfig:134]: I'm fine with it, but [this rule is already set to `None`](https://github.com/Lombiq/.NET-Analyzers/blob/9a30791b5c1074abbf106a6e9eca9a58e90aa669/general.ruleset#L90) in the _general.ruleset_, so what gives?

Comment (0liver) [reply .editorconfig:134]: Well, guess what?

It didn't do the trick 🤓

File: orchardcore.ruleset (hunk 2 above)
Comment (sarahelsaig) [reply orchardcore.ruleset:11]: I don't see that warning in Rider at all, though I may have disabled it locally.

**Review by 0liver — APPROVED** [review]

**Pull request merged**